cogs/rcon.py

```diff
@@ -0,0 +1,262 @@
+                if command == "status":
+                    await ctx.send("```\n" + response.split('#', 1)[1] + "\n```")
+                else:
+                #for everything else
+                    if response == "":
+                        await ctx.send("Command sent.")
+                    else:
+                        await ctx.send("```\n" + response + "\n```")
+            #usmvm
+            elif server == "usmvm":
+                command = " ".join(commands)
+                response = await run_rcon(server, command)
+
+                #for status
+                if command == "status":
+                    await ctx.send("```\n" + response.split('#', 1)[1] + "\n```")
+                else:
+                #for everything else
+                    if response == "":
+                        await ctx.send("Command sent.")
+                    else:
+                        await ctx.send("```\n" + response + "\n```")
+            #rare edge case
+            else:
+                await ctx.send("How did it reach this part?")
+        #wrong server
+        else:
+            await ctx.send("Invalid server. See `?servers`.")
+
+    #status test command TODO
+    @command()
+    @has_any_role('Staff', 'Server Mods', 'Senior Staff', 'Fub')
+    async def status(self, ctx, server):
+        await ctx.trigger_typing()
+        if server == "us":
+            await ctx.send("us imp")
+        elif server == "us-mvm":
+            await ctx.send("us mvm")
+        elif server == "eu":
+            await ctx.send("eu imp")
+            
+            #server info
+            euserver = server_info(config.rcon.euip, config.rcon.imp)
+            await ctx.send(euserver)
+
+            #player info???
+            euserverplayers = server_players(config.rcon.euip, config.rcon.imp)
+            await ctx.send(euserverplayers)
+
+            #player info???
+            #eurules = get_nextmap('eu.tf2maps.net', 27015)
+            #await ctx.send(eurules)
+            #print(eurules)
+
+        elif server == "eu-mvm":
+            await ctx.send("eu mvm")
+        else:
+            await ctx.send("no server found")
+
+    #nextmap
+    @command()
+    @has_any_role('Staff', 'Server Mods', 'Senior Staff', 'Fub')
+    async def nextmap(self, ctx, server):
+        await ctx.trigger_typing()
+        if server == "us":
+
+            nextmap = get_nextmap('us.tf2maps.net', 27015)
+            await ctx.send(nextmap)
+
+        elif server == "us-mvm":
+
+            nextmap = get_nextmap('us.tf2maps.net', 27016)
+            await ctx.send(nextmap)
+
+        elif server == "eu":
+
+            nextmap = get_nextmap('eu.tf2maps.net', 27015)
+            await ctx.send(nextmap)
+
+        elif server == "eu-mvm":
+
+            nextmap = get_nextmap('eu.tf2maps.net', 27016)
+            await ctx.send(nextmap)
+        
+        else:
+            await ctx.send("no server found")
+
+#get server info via a2s
+def server_info(host, port):
```

Comment:
these 2 functions prob dont need to be in a function. these calls are just doing the exact same thing as the a2s thing

Reply:
server_info and the player one was from old code I think. Removed it.